spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/GCPProperties.java

```diff
@@ -27,7 +28,7 @@
 
 	private String projectId;
 
-	private String jsonKeyFile;
+	private Resource credentialKeyFile;
```

Comment:
@meltsufin Anything that we believe needs to be configurable via Boot's abstraction I think should be on the GCPProperties. Things like : projectId, Region, credentials.

If there's a way to obtain the projectId from the credentials (I think I asked that before and got a negative reply)

Then we could make it a default value being the one coming from the credentials.

The sole reason of the Properties file is to be a placeholder to carry config options. Nothing more.

AutoConfiguration classes than rely on that to create the beans as well as to make decision on Conditional. There's a `ConditionalOnProperty` that can assist on activation of beans depending on properties being set or not.